### DISCUSSION TITLE
Queries not fetched on background click in V5

### DISCUSSION THREAD

**Discussion by lveillard** [TanStack/query#6200] [Q&A] [answered]
One thing that has changed after we migrated to V5 is that suddenly clicking the background no longer re-fetches the queries. 

Is this done on purpose? It keeps refetching on window focus.

Before:
- After some seconds, any click on the current window even without changing the window triggers refetch of current querie

Now:
- Queries are not refetched unless we change the window and come back to it

**Comment by TkDodo** [comment, marked as answer]
> Is this done on purpose?

yes, and it's documented in the v5 migration guide:
- https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#window-focus-refetching-no-longer-listens-to-the-focus-event

**Reply by lveillard** [reply]
I see thanks, read it twice and did not find that one.

Is there a way to force it back or it's completely removed? Is one of the reasons we use react-query, data is contantly refetched on user interaction and that was great

**Reply by TkDodo** [reply]
it will still be fetched von user interaction, unless you have multiple monitors with two windows that receive focus without triggering a `visibilitychange`. The root use-case is "user comes back to the page", which is when e.g. switching between tabs.

But yes, you can setup your own event listener with `focusManager.setEventListener`:

https://tanstack.com/query/v5/docs/react/reference/focusManager#focusmanagerseteventlistener

you can do `visbilitychange` + `focus` event if you want
